### ISSUE TITLE
Plugin is not linting.

### ISSUE THREAD

**Issue by bk-dev** [vuejs/eslint-plugin-vue#336]
**Tell us about your environment**

* **ESLint Version: 4.10.0**
* **eslint-plugin-vue Version: 4.0.0**
* **Node Version: 6.9.5**

**Please show your full configuration:**

```
module.exports = {
    root: true,
    env: {
        browser: true
    },
    extends: [
        "airbnb-base",
        "plugin:vue/recommended"
    ],
    plugins: [
        "html"
    ],
    rules: {
        "indent": [2, 4, { "SwitchCase": 1 }],
        "comma-dangle": [2, "never"],
        "no-param-reassign": [0, "never"],
        "max-len": [0, 200],
        "import/no-extraneous-dependencies": [0],
        "import/extensions": [0],
        "import/no-unresolved": [0],
        "import/first": [0],
        "linebreak-style": [0],
        "no-plusplus": [0],
        'no-debugger': process.env.NODE_ENV === 'production' ? 2 : 0,
        'no-console': process.env.NODE_ENV === 'production' ? 2 : 0
    }
};
```

**What did you do? Please include the actual source code causing the issue.**
I have dozens of components (i.e. .vue files) within my project that are already getting linted properly (i.e. javascript code only).

**What did you expect to happen?**
Vue linting errors to be reported.

**What actually happened? Please include the actual, raw output from ESLint.**
No errors reported.

**Comment by bk-dev** [comment]
I had to remove 'html' from the plugin array for this to work leaving just 'vue' as the only item. Not sure why that is. I read somewhere that eslint-plugin-html was a dependency for eslint-plugin-vue, so it's a little strange.

**Comment by mysticatea** [comment]
This plugin requires the whole code of `.vue` file to lint `<template>` and `<script>`, but `eslint-plugin-html` extracts the code inside of `<script>` element, and so it makes this plugin to no be able to find `<template>` and `<script>`.
